test/Counter.test.ts

```diff
@@ -16,31 +16,34 @@ describe("Counter", () => {
     counter = (await ethers.getContractAt(
       "Counter",
       counterAddress
-    )) as Counter;
+    )) as unknown as Counter;
   });
 
   it("increment", async () => {
     const [deployer] = await ethers.getSigners();
 
-    const maxFee = ethers.utils.parseEther("1000");
-    const kmon = (await ethers.getContractAt("IERC20", KMON_TOKEN)) as IERC20;
+    const maxFee = ethers.parseEther("1000");
+    const kmon = (await ethers.getContractAt(
+      "IERC20",
+      KMON_TOKEN
+    )) as unknown as IERC20;
 
     // transfer from safe to deployer
     const kmonSafe = await ethers.getImpersonatedSigner(KMON_SAFE);
     await kmon.connect(kmonSafe).transfer(deployer.address, maxFee);
 
     // deployer approve counter to spend fee
-    await kmon.connect(deployer).approve(counter.address, maxFee);
+    await kmon.connect(deployer).approve(counter.getAddress(), maxFee);
```

Comment:
replace `getAddress` with `target`.
also `getAddress` returns a promise, make sure to await.